WooCommerce/WooCommerceTests/Tools/InfiniteScroll/ScrollWatcherTests.swift

```diff
@@ -0,0 +1,34 @@
+import XCTest
+@testable import WooCommerce
+
+final class ScrollWatcherTests: XCTestCase {
+    private var cancellable: ObservationToken?
+
+    func test_scroll_trigger_is_emitted_only_after_scrolling_to_above_threshold() {
+        // Arrange
+        let frame = CGRect(origin: .zero, size: CGSize(width: 320, height: 100))
+        let tableView = UITableView(frame: frame, style: .grouped)
+        let contentSize = CGSize(width: 320, height: 1100)
+        tableView.contentSize = contentSize
+        let threshold = 0.8
+        let scrollWatcher = ScrollWatcher(scrollPositionThreshold: threshold)
+        scrollWatcher.startObservingScrollPosition(tableView: tableView)
+        var triggeredScrollPositions = [Double]()
+        cancellable = scrollWatcher.scrollTrigger.subscribe { scrollPosition in
+            triggeredScrollPositions.append(scrollPosition)
+        }
```

Comment:
To avoid the ivar, we can also use `addTeardownBlock`

```suggestion
        let cancellable = scrollWatcher.scrollTrigger.subscribe { scrollPosition in
            triggeredScrollPositions.append(scrollPosition)
        }

        addTeardownBlock {
            cancellable.cancel()
        }
```

Reply:
TIL'ed `addTeardownBlock`! in case we add more test cases, it might be nice to have the `cancellable` as a property. I added the `tearDown` override to cancel the observation token in https://github.com/woocommerce/woocommerce-ios/pull/2747/commits/2f42e68a3b2bbbec9a582a2341dbbcfbc15f9e6d